### DISCUSSION TITLE
Is it possible to activate a virtual environment with Python 3.11.2 via SSH on Linux?

### DISCUSSION THREAD

**Discussion by ebalvis** [pyscripter/pyscripter#1353] [Q&A] [answered]
Hi,

I’m working on a Linux server with Python 3.11.2 (compiled with GCC 12.2.0) and accessing it via SSH. I have a question: is it possible to activate a Python virtual environment (venv) in this setup?

Key details:

Python 3.11.2 is already installed and working properly.
I access the server remotely via SSH.
I’d like to know if I can activate and use a virtual environment in this context, and if so, what are the correct steps to follow?
Any advice or guidance would be greatly appreciated. Thank you!

**Comment by pyscripter** [comment, marked as answer]
You first create the virtual environment on the server:

1. Install venv if it is not already installed. e.g.
`$ sudo apt install python3.10-venv`
2. Create the venv e.g.
`$ python3 -m venv /home/username/python/venv`

 In PyScriprer :
1. Create a new SSH server with "Command to python executable":  /home/username/python/venv/bin/python3
2. Select the new server to use it.
